compiler/damlc/daml-prim-src/GHC/Num.daml

```diff
@@ -148,6 +148,33 @@ instance Signed IF_NUMERIC((Numeric n), Decimal) where
     signum x = if x == aunit then aunit else if x <= aunit then negate munit else munit
     abs x = if x <= aunit then negate x else x
 
+#ifdef DAML_BIGNUMERIC
+
+instance Additive BigNumeric where
+    (+) = primitive @"BEAddBigNumeric"
+    (-) = primitive @"BESubBigNumeric"
+    negate x = aunit - x
+    aunit = (primitive @"BEFromNumericBigNumeric" : Numeric 0 -> BigNumeric) (0.0 : Numeric 0)
+
+instance Multiplicative BigNumeric where
+    (*) = primitive @"BEMulBigNumeric"
+    munit = (primitive @"BEFromNumericBigNumeric" : Numeric 0 -> BigNumeric) (1.0 : Numeric 0)
+    x ^ n
+        | n == 0 = munit
+        | n == 2 = x * x
+        | n < 0 = (primitive @"BEDivBigNumeric" : Int -> RoundingMode -> BigNumeric -> BigNumeric -> BigNumeric)
+                    1024 (primitive @"BERoundingUnnecessary" : RoundingMode) munit (x ^ (negate n))
```

Comment:
The `(^)` operator is mandatory in Multiplicative. It seems less evil to use a division (with no rounding) than to return a 0 for negative powers. But at this point we may as well have an instance for Divisible and Fractional...

Reply:
As mentioned in slack, I think I’d be tempted to make this crash for negative powers but either way perfectly fine to leave it like this for now. Maybe add a TODO to revisit this?